src/compiler/utilities.ts

```diff
@@ -1913,6 +1914,12 @@ namespace ts {
         return token >= SyntaxKind.FirstAssignment && token <= SyntaxKind.LastAssignment;
     }
 
+    export function isClassExtendsExpressionWithTypeArguments(node: Node): boolean {

```

Comment:
Can you reword this name? It sounds weird in English to have "extends" modify "expression with type arguments".

Actually, if you make this take `ExpressionWithTypeArguments` instead of `Node`, you can just call it `isInClassExtendsClause`.